ncm-sudo/src/main/perl/sudo.pm

```diff
@@ -197,8 +199,11 @@ sub generate_privilege_lines {
         my $ln = $info{PRIVILEGE_USER()}->getValue;
         $ln .= "\t" . $info{PRIVILEGE_HOST()}->getValue;
         $ln .= "= (". $info{PRIVILEGE_RUNAS()}->getValue . ")\t";
-        $ln .= $info{PRIVILEGE_OPTS()}->getValue . "\t"
-        if defined $info{PRIVILEGE_OPTS()};
+	if (defined($info{PRIVILEGE_OPTS()})) {
+		$ln .= $info{PRIVILEGE_OPTS()}->getValue;
+		$ln .= ':' if $ln !~ /:$/;
```

Comment:
Yeah, this is enforcing it by the component, but only if it isn't set via the templates. And it makes the colon optional in the templates.

The code used to always add a colon at the end (changed in 31333b8). This means if we upgrade the schema and change the templates before the component is upgraded, we will have two colons at the end of the line, and the sudoers file will not apply.

If we upgrade the component before the schema, we will have no colon after the option, and again sudoers will not apply.

Because we cannot upgrade both at the same time, it makes upgrading to this latest upstream in its current state quite awkward. This change should allow either way to work.